src/main/java/org/dcsa/bkg/controller/BKGController.java

```diff
@@ -27,17 +27,17 @@
 
   @PostMapping
   @ResponseStatus(HttpStatus.ACCEPTED)
-  public Mono<BookingTO> createBooking(@Valid @RequestBody BookingTO bookingRequest) {
-    return bookingService.createBooking(bookingRequest);
+  public Mono<BookingResponseTO> createBooking(@Valid @RequestBody BookingTO bookingRequest) {
+    return bookingService.createBooking(bookingRequest).flatMap(bookingService::toBookingResponseTO);
   }
 
   @PutMapping("/{carrierBookingRequestReference}")
   @ResponseStatus(HttpStatus.OK)
-  public Mono<BookingTO> updateBookingByReference(
+  public Mono<BookingResponseTO> updateBookingByReference(
       @PathVariable @Size(max = 100) String carrierBookingRequestReference,
       @Valid @RequestBody BookingTO bookingRequest) {
     return bookingService.updateBookingByReferenceCarrierBookingRequestReference(
-        carrierBookingRequestReference, bookingRequest);
+        carrierBookingRequestReference, bookingRequest).flatMap(bookingService::toBookingResponseTO);
```

Comment:
I feel like the setup could be simpler. Now we first create the booking transform the response to the bookingTO and transform the bookingTo to the bookingResponseTO. Whereas is we directly mapt to the bookingResponseTO in the BookingService we can simplify a lot of the steps

Reply:
That was my initial approach as well, but it breaks a lot of implementation test cases and Frazer suggested we just transform BookingTO to BookingResponseTO instead.